src/data/position.rs

```diff
@@ -4,14 +4,16 @@
 #[macro_use]
 use corollary_support::*;
 
+use std::sync::Arc;
+
 // TODO use a deque
 use data::r_list::Reversed;
 
 #[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd, Hash)]
 pub enum Position {
     Position {
         offset: isize,
-        file: String,
+        file: Arc<String>,
```

Comment:
That was before I removed the parse-within-thread. With that, the AST was returned to the main thread, so Position needed to be `Send`.